Sources/LanguageServerProtocol/SupportTypes/Diagnostic.swift

```diff
@@ -111,11 +126,48 @@ extension DiagnosticCode: Codable {
   }
 }
 
-extension DiagnosticCode: CustomStringConvertible {
+extension DiagnosticCode: Codable {
+  enum ComplexCodingKeys: CodingKey {
+    case value
+    case target
+  }
+
+  public init(from decoder: Decoder) throws {
+    if let simpleCodeValue = try? NumberOrString(from: decoder) {
+      self.value = simpleCodeValue
+    } else {
+      let container = try decoder.container(keyedBy: ComplexCodingKeys.self)
+      self.value = try container.decode(NumberOrString.self, forKey: .value)
+      self.target = try? container.decode(DocumentURI.self, forKey: .target)
```

Comment:
`try container.decodeIfPresent` will let us skip a non-existent key but still propagate other deserialization errors (e.g. it's the wrong type).